lib/Zip/ioapi.cpp

```diff
@@ -127,8 +130,16 @@ fopen64_file_func(voidpf opaque, const void* filename, int mode) {
   else if (mode & ZLIB_FILEFUNC_MODE_CREATE)
     mode_fopen = "wb";
 
-  if ((filename != NULL) && (mode_fopen != NULL))
+  if ((filename != NULL) && (mode_fopen != NULL)){
     file = FOPEN_FUNC((char const*)filename, mode_fopen);
+  }
+#ifndef _MSC_VER
+  if(file == NULL) {
+    auto out = std::unique_ptr<char[]>(new char[256]);
+    ::strerror_r(errno,out.get(),256);
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_SYS_ERROR, out.get());
```

Comment:
From the manual:
```
       The XSI-compliant strerror_r() is preferred for portable  applications.
       It  returns  the error string in the user-supplied buffer buf of length
       buflen.

       The GNU-specific strerror_r() returns a pointer to a string  containing
       the  error  message.  This may be either a pointer to a string that the
       function stores in buf, or a pointer to some (immutable) static  string
       (in which case buf is unused).  If the function stores a string in buf,
       then at most buflen bytes are stored (the string may  be  truncated  if
       buflen is too small and errnum is unknown). 
```
That means `strerror_r` may actually not store something inside `out`. So we should always use the return value of the function for capturing the message.